addons/portal/static/src/scss/portal.scss

```diff
@@ -362,8 +362,15 @@ img, .media_iframe_video, .o_image {
             @include o-position-sticky($top: $spacer * 5);
             z-index: 1;
         }
-    }
-}
+        h2 {
```

Comment:
Be more specific => `h2:has(.oe_currency_value) {`
Even though it's not the case by default, other H2 headings added through customer customizations might be present in the `o_portal_sidebar_content`.